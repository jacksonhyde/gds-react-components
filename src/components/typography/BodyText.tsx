import React, { HTMLProps } from 'react';
import classNames from 'classnames';

const BodyText: React.FC<HTMLProps<HTMLParagraphElement>> = ({ className, ...rest }) => (
  <p className={classNames('govuk-body', className)} {...rest} />
);

export default BodyText;
