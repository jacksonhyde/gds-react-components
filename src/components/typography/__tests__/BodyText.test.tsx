import React from 'react';
import { shallow } from 'enzyme';
import BodyText from '../BodyText';

describe('BodyText', () => {
  it('matches snapshot', () => {
    const element = shallow(<BodyText>Text</BodyText>);
    expect(element.text()).toBe('Text');
    expect(element).toMatchSnapshot();
    expect(element.hasClass('govuk-body')).toBeTruthy();
    element.unmount();
  });
});
