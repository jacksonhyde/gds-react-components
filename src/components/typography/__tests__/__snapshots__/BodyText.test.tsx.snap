// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyText matches snapshot 1`] = `
<p
  className="govuk-body"
>
  Text
</p>
`;
